ext/soap/php_encoding.c

```diff
@@ -1380,8 +1394,24 @@ static zval *to_zval_object_ex(zval *ret, encodeTypePtr type, xmlNodePtr data, z
 	} else if (SOAP_GLOBAL(class_map) && type->type_str) {
 		zval              *classname;
 		zend_class_entry  *tmp;
-
-		if ((classname = zend_hash_str_find_deref(SOAP_GLOBAL(class_map), type->type_str, strlen(type->type_str))) != NULL &&
+        classname = zend_hash_str_find_deref(SOAP_GLOBAL(class_map), type->type_str, strlen(type->type_str));
+        if(classname == NULL){
+            if (type->ns) {
+                int ns_len = strlen(type->ns);
```

Comment:
Same overflow remark as earlier. In fact, this code may be simplified by using `zend_strpprintf` which works like (s)printf and returns a `zend_string *`. That pointer may then be used with `zend_hash_find_deref` which will also be a bit faster than `zend_hash_str_find_deref`.